### PR TITLE
tools: fix set namespace in pd-ctl

### LINKED ISSUE #1681: `pd-ctl config set namespace` should raise error when the namespace does not exist?
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
» config set namespace ts1 leader-schedule-limit 10
Success!
» config show namespace ts1
[404] "invalid namespace Name ts1, not found"

2. What did you expect to see?
» config set namespace ts1 leader-schedule-limit 10
[404] "invalid namespace Name ts1, not found"

3. What did you see instead?
» config set namespace ts1 leader-schedule-limit 10
Success!

4. What version of PD are you using (`pd-server -V`)?
```bash
$ bin/pd-ctl -V
Release Version: v4.0.0-alpha-26-g96e25880
Git Commit Hash: 96e258805daee08029f6e5ba885c8861e33d9d73
Git Branch: master
UTC Build Time:  2019-08-14 08:51:44

$ bin/pd-server -V
Release Version: v4.0.0-alpha-26-g96e25880
Git Commit Hash: 96e258805daee08029f6e5ba885c8861e33d9d73
Git Branch: master
UTC Build Time:  2019-08-14 08:51:42
```

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve? 
When the status of the response is not OK, it should return an error. Closes #1681. 

### What is changed and how it works?
After this PR:
```
$ ./pd-ctl -u http://127.0.0.1:2379 -i
>> config set namespace ts1 leader-schedule-limit 10
Failed to set namespace: ts1 error: [404] "invalid namespace Name ts1, not found"
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: tools/pd-ctl/pdctl/command/global.go [hunk 1]
```diff
@@ -174,23 +174,29 @@ func postJSON(cmd *cobra.Command, prefix string, input map[string]interface{}) {
...
 		defer r.Body.Close()
 		if r.StatusCode != http.StatusOK {
-			if err := printResponseError(r); err != nil {
-				cmd.Println(err)
+			msg, err = ioutil.ReadAll(r.Body)
+			if err != nil {
+				return err
 			}
-			return nil
+			return errors.Errorf("[%d] %s", r.StatusCode, msg)
```
Comment (shafreeck) [inline tools/pd-ctl/pdctl/command/global.go]: I think we should only try next endpoint on network error. If the request was returned with a status code is not OK, there may be a problem of client, not server.

Comment (rleungx) [reply tools/pd-ctl/pdctl/command/global.go]: But the error should be returned here anyway.

Comment (rleungx) [reply tools/pd-ctl/pdctl/command/global.go]: But the error should be returned here anyway.

**Pull request merged**